samcli/lib/deploy/deployer.py

```diff
@@ -444,7 +445,9 @@ def _is_root_stack_event(event: Dict) -> bool:
     def _check_stack_not_in_progress(status: str) -> bool:
         return "IN_PROGRESS" not in status
 
-    def wait_for_execute(self, stack_name: str, stack_operation: str, disable_rollback: bool) -> None:
+    def wait_for_execute(
+        self, stack_name: str, stack_operation: str, disable_rollback: bool, execution_time: float
```

Comment:
Should we default `execution_time` to `time.time()? This make the method a bit more intuitive to use - If I don't need to check against a specific timestamp, I just don't provide this argument.

```suggestion
        self, stack_name: str, stack_operation: str, disable_rollback: bool, execution_time: float = time.time()
```